lib/substitutions.js

```diff
@@ -20,6 +22,22 @@ add("&lt;", "<");
 add("&gt;", ">");
 add("&amp;", "&"); // &amp; must be after all replacements involving &s.
 
+var emojiIndex = (function indexEmoji() {
+    var emojiJson = fs.readFileSync("./external/emoji-data/emoji.json");
+    var emoji = JSON.parse(emojiJson);
+    var index = {};
+    for (var i = 0; i < emoji.length; ++i) {
+        var e = emoji[i];
+        for (var j = 0; j < e.short_names.length; ++j) {
+            if (e["unified"].indexOf("-") < 0) {
+                var char = String.fromCodePoint(parseInt(e["unified"], 16));
+                index[":" + e.short_names[j] + ":"] = char;
+            }
+        }
+    }
+    return index;
+})();
```

Comment:
Why are we doing this instead of just assigning directly to `emojiIndex`? Also, it needlessly obfuscates the code making this a named function `indexEmoji` which we never actually use anywhere.